privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -5,61 +5,49 @@
 from privacy_evaluator.utils.trainer import trainer
 from privacy_evaluator.models.torch.fc_neural_net import FCNeuralNet
 
-import math
 import numpy as np
 import torch
 import tensorflow as tf
 from sklearn.svm import SVC
 from sklearn.model_selection import train_test_split
+from sklearn.utils import shuffle
 from art.attacks.evasion import FastGradientMethod
 from art.estimators.classification import SklearnClassifier
-from typing import Tuple, Any, Dict, List
+from typing import Tuple, Any, Dict, List, Optional
 
 
 class PropertyInferenceAttack(Attack):
-    def __init__(self, target_model: Classifier, dataset: Tuple[np.ndarray,np.ndarray]):
+    def __init__(self, target_model: Classifier, dataset: Tuple[np.ndarray, np.ndarray]):
         """
         Initialize the Property Inference Attack Class.
         :param target_model: the target model to be attacked
         :param dataset: dataset for training of shadow classifiers, test_data from dataset with concatenation [test_features, test_labels]
         """
-        self.dataset=dataset
-        # count of shadow training sets, must be eval
-        self.amount_sets = 6
-        self.input_shape =  self.dataset[0].shape #[32, 32, 3]
+        self.dataset = dataset
+        # count of shadow training sets, must be even
+        self.amount_sets = 2 #TODO change to higher number
```

Comment:
Things that have hard-coded values and TODO statements cannot be merged! This is a KO criterion.

Reply:
I deleted the TODO statement.